### PR TITLE
[DROOLS-3290] Bump scesim file version to 1.1, added automatic migration support (in memory for now)

### PR DESCRIPTION
**Pull request by danielezonca**
Define automatic migration mechanism. It is in memory for now to be compatible with both internal WB and external maven runner.

@kkufova @Rikkola @gitgabrio

### CODE REVIEW

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/util/InMemoryMigrationStrategy.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package org.drools.workbench.screens.scenariosimulation.backend.server.util;
+
+import java.util.function.Function;
+
+public class InMemoryMigrationStrategy implements MigrationStrategy {
+
+    @Override
+    public Function<String, String> from1_0to1_1() {
```
Comment (gitgabrio) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/util/InMemoryMigrationStrategy.java]: I think this method name should be changed to "migrate" or something anyway "agnostic" of the actual migration level.

**Comment by danielezonca** [comment]
@gitgabrio 
Yes my idea is to have a specific method for migration for each version (1.0 -> 1.1, 1.1 -> 1.2, etc). I agree that this is not the best if we will have a lot of migrations steps but this has the advantage for now that it is statically defined and it is impossible to forget to implement any step of the migration strategy.
I would like to keep this approach and move to a generic list of ordered migration steps only if needed (aka > 10 steps)

**Comment by danielezonca** [comment]
@manstis 
Can you please squash and merge?

**Comment by Rikkola** [comment]
@gitgabrio @danielezonca It is unlikely that we need that many steps. Adding new items does not need migration, only if something is removed/changed. And I think step by step migration from one version to the next is the only way we can stay sane.

**Pull request merged**